### DISCUSSION TITLE
Will this framework benefit from using virtual threads?

### DISCUSSION THREAD

**Discussion by SentryMan** [activej/activej#291] [General]
JDK 21 is right around the corner, so I'm wondering if there will be any integrations with it.

**Comment by eduard-vasinskyi** [comment]
Hi, @SentryMan

We are not currently considering virtual threads for ActiveJ. 

ActiveJ adopts a multireactor pattern. Each reactor runs in a dedicated thread. As said in [the virtual threads adoption guide](https://docs.oracle.com/en/java/javase/21/core/virtual-threads.html#GUID-8AEDDBE6-F783-4D77-8786-AC5A79F517C0):

> Programs or frameworks written in other styles that don't dedicate a thread per task should not expect to see a significant benefit from virtual threads.


But we will look closely at this new feature, maybe we could utilize it in other ActiveJ modules.
